src/main/perf/LineFileDocs.java

```diff
@@ -139,6 +136,7 @@ private void readDocs() throws Exception {
       byte[] headerBytes = new byte[8];
       ByteBuffer header = ByteBuffer.wrap(headerBytes);
       header.order(ByteOrder.LITTLE_ENDIAN);
+      int totalDocCount = 0;  // hopefully this won't overflow
```

Comment:
It shouldn't -- Lucene cannot index more than `Integer.MAX_VALUE` docs since it uses `int docid`.

Reply:
Hmmm well there's still chance? Since that's a per segment id?

Reply:
Yeah, that is a good point.

However, `IndexWriter` tries to protect this -- it sums the total doc count across all segments, even including incoming from `addIndexes` calls, and prevent the whole index from growing over the `int` limit.  It is sort of a best-effort thing though ;)  So there is STILL a chance.  Turtles all the way down...